awx/main/managers.py

```diff
@@ -170,9 +171,7 @@ def register(self, uuid=None, hostname=None, ip_address=None, node_type='hybrid'
             }
             if defaults is not None:
                 create_defaults.update(defaults)
-            uuid_option = {}
-            if uuid is not None:
-                uuid_option = {'uuid': uuid}
+            uuid_option = {'uuid': node_uuid if node_uuid is not None else uuid.uuid4()}
```

Comment:
@kdelee we pass the node_uuid as the uuid if exists, if not, we generate a new one